### ISSUE TITLE
Fix Eclipse support

### ISSUE THREAD

**Issue by tpetracca** [palantir/atlasdb#1698]
I have to do all sorts of manual classpath editing to get things working properly in eclipse.
Looks like we changed some of the immutables generation that now requires that eclipse depend on `build/classes/[main|test]` as a source folder depending on the project. So far I've had to do this for:
- atlasdb-config
- atlasdb-impl-shared
- atlasdb-persistent-lock-api

Others I needed to add `generated_src` and/or `generated_testSrc` to the build path:
- atlasdb-cassandra
- atlasdb-config

I still haven't quite gotten it right yet.

**Comment by clockfort** [comment]
@twilson-palantir has also had similar eclipse issues when attempting to interact with Atlas as of late

**Comment by SerialVelocity** [comment]
This may be something to do with us half using the old intellij logic and half using the baseline logic.

**Comment by rhero** [comment]
@tpetracca This is really weird, but we're realistically not going to fix this as nobody on the team uses Eclipse. Feel free to put up PRs that incrementally fix the issues you hit.

**Comment by tpetracca** [comment]
@rhero agreed it's not a high priority, but leaving this open as it 100% has to get addressed long term.

So #1694 fixed the good majority of my errors. Thanks @hsaraogi!

The remaining errors seem to all be related to our attempting to have `testCompile` dependencies on `testArtifacts` of other projects. For example, in `atlasdb-cassandra/build.gradle` we see
```
dependencies {
    testCompile project(path: ":atlasdb-client", configuration: "testArtifacts")
}
```

This line ends up causing a build path issue in eclipse, due to it looking for a directory that indeed does not exist (even after you run `./gradlew atlasdb-client:build`
> Project 'atlasdb-cassandra' is missing required library: '\<path-to-my-local-atlasdb-repo\>/atlasdb-client/build/resources/test'

**Comment by rhero** [comment]
@hsaraogi will take a look at this for a few minutes today, but if we can't fix it quickly, we'll throw this back into the backlog.

**Comment by gsheasby** [comment]
Should explicitly remove eclipse support instead.

**Comment by tpetracca** [comment]
Fixed by #1993
